File: dbfit-2.x/dbfit-fitnesse/src/main/java/dbfit/fixture/Clean.java
package dbfit.fixture;

import java.math.BigDecimal;
import java.sql.SQLException;

import org.dbfit.core.DBEnvironment;
import org.dbfit.core.DbEnvironmentFactory;
import fit.Parse;

public class Clean extends fit.ColumnFixture {
    private DBEnvironment environment;
	public Clean(DBEnvironment environment) {
		this.environment = environment;
	}
    public Clean()
    {
        this.environment = DbEnvironmentFactory.getDefaultEnvironment();
    }
    public String table;
    public String columnName;
    public BigDecimal[] ids;
    public String[] keys;
    public String where=null;
    private String getIDCSV()
    {
        StringBuilder sb = new StringBuilder();
        String comma = "";
        for (BigDecimal x :ids){
            sb.append(comma);
            sb.append(x.toString());
            comma = ", ";
        }
        return sb.toString();
    }
    private String getKeyCSV() 
    {
        StringBuilder sb = new StringBuilder();
        String comma = "";
        for (String x : keys)
        {
            sb.append(comma);
            sb.append("'");
            sb.append(x.toString());
            sb.append("'");
            comma = ", ";
        }
        return sb.toString();
    }

    private boolean hadRowOperation=false;		
	public boolean clean() throws SQLException{
		String s=  "Delete from " + table +(where!=null?(" where "+where):"");
		environment.createStatementWithBoundFixtureSymbols(s).execute();
		return true;
	}
    public boolean DeleteRowsForIDs() throws SQLException
    {
        String s=    "Delete from " + table + " where "+columnName +" in ("
            + getIDCSV()+") "+(where != null ? " and " + where : "");
		environment.createStatementWithBoundFixtureSymbols(s).execute();
		hadRowOperation=true;
		return true;
    }
    public boolean DeleteRowsForKeys() throws SQLException
    {
    	String s=   "Delete from " + table + " where " + columnName + " in ("
            + getKeyCSV() + ") " + (where != null ? " and " + where : "");
    	environment.createStatementWithBoundFixtureSymbols(s).execute();
    	hadRowOperation=true;
    	return true;
    }
	public void doRow(Parse row) {
		hadRowOperation = false;
		super.doRow(row);
		if (!hadRowOperation) {
			try{
				clean();
			}
			catch (SQLException sqle){
				exception(row,sqle);	
			}
		}
	}
}

File: dbfit-dotnet/core/src/fixture/Clean.cs
/// Copyright (C) Gojko Adzic 2006-2008 http://gojko.net
/// Released under GNU GPL 2.0
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Common;
using System.Text;

using fit;


namespace dbfit.fixture {
    /// <summary>
    /// Deprecated fixture that can be used to clean up the database. Kept only for
    /// legacy compatibility. Use Execute fixture with a Delete command instead of this.
    /// </summary>
	public class Clean:fit.ColumnFixture {

        private IDbEnvironment environment;
		public Clean(IDbEnvironment environment) {
			this.environment = environment;
		}
        public Clean()
        {
            this.environment = DbEnvironmentFactory.DefaultEnvironment;
        }
        public String table;
        public String columnName;
        public Decimal[] ids;
        public String[] keys;
        public string where=null;
        private string getIDCSV()
        {
            StringBuilder sb = new StringBuilder();
            String comma = "";
            foreach (decimal x in ids)
            {
                sb.Append(comma);
                sb.Append(x.ToString());
                comma = ", ";
            }
            return sb.ToString();
        }
        private string getKeyCSV()
        {
            StringBuilder sb = new StringBuilder();
            String comma = "";
            foreach (String x in keys)
            {
                sb.Append(comma);
                sb.Append("'");
                sb.Append(x.ToString());
                sb.Append("'");
                comma = ", ";
            }
            return sb.ToString();
        }

        private bool hadRowOperation=false;		
		public bool clean() {
            DbCommand command = environment.CreateCommand(
                "Delete from " + table +(where!=null?" where "+where:""), CommandType.Text);
            command.ExecuteNonQuery();
			return true;
		}
        public bool DeleteRowsForIDs()
        {
            DbCommand command = environment.CreateCommand(
                "Delete from " + table + " where "+columnName +" in ("
                + getIDCSV()+") "+(where != null ? " and " + where : ""), CommandType.Text);
            command.ExecuteNonQuery();
        	hadRowOperation=true;
            return true;
        }
        public bool DeleteRowsForKeys()
        {
            DbCommand command = environment.CreateCommand(
                "Delete from " + table + " where " + columnName + " in ("
                + getKeyCSV() + ") " + (where != null ? " and " + where : ""), CommandType.Text);
            command.ExecuteNonQuery();
        	hadRowOperation=true;
        	return true;
        }
		public override void DoRow(Parse row) {
			hadRowOperation = false;
			base.DoRow(row);
			if (!hadRowOperation) {
				clean();
			}
		}
	}
}